api/app.go

```diff
@@ -332,7 +375,97 @@ func (app *App) Start() error {
 				zap.Int("graceperiod", graceperiod))
 		})
 		time.Sleep(time.Duration(graceperiod) * time.Millisecond)
+	case err := <-errch:
+		return err
+	case <-stopped:
 	}
 	log.I(l, "app stopped")
 	return nil
 }
+
+func (app *App) startGRPCServer(lis net.Listener, errch chan<- error) {
+	var basicAuthInterceptor grpc.UnaryServerInterceptor
+
+	basicAuthUser := app.Config.GetString("basicauth.username")
+	if basicAuthUser == "" {
+		basicAuthInterceptor = grpc.UnaryServerInterceptor(app.noAuthMiddleware)
+	} else {
+		basicAuthInterceptor = grpc_auth.UnaryServerInterceptor(app.basicAuthMiddleware)
+	}
+
+	app.grpcServer = grpc.NewServer(grpc.UnaryInterceptor(
+		grpc_middleware.ChainUnaryServer(
+			basicAuthInterceptor,
+			grpc.UnaryServerInterceptor(app.loggerMiddleware),
+			grpc.UnaryServerInterceptor(app.recoveryMiddleware),
+			grpc.UnaryServerInterceptor(app.responseTimeMetricsMiddleware),
+			grpc.UnaryServerInterceptor(app.sentryMiddleware),
+			grpc.UnaryServerInterceptor(app.newRelicMiddleware),
+		),
+	))
+	api.RegisterPodiumServer(app.grpcServer, app)
+
+	app.grpcReady <- true
+	if err := app.grpcServer.Serve(lis); err != nil {
+		errch <- fmt.Errorf("error trying to serve with grpc server: %v", err)
+	}
+}
+
+func (app *App) startHTTPServer(lis net.Listener, errch chan<- error) {
+	gatewayMux := runtime.NewServeMux(runtime.WithMarshalerOption(runtime.MIMEWildcard,
```

Comment:
nit: try to line break at a consistent level. Eg:

```golang
gatewayMux := runtime.NewServeMux(
    runtime.WithMarshalerOption(runtime.MIMEWildcard, &runtime.JSONPb{EmitDefaults: true}))
```

or

```golang
gatewayMux := runtime.NewServeMux(runtime.WithMarshalerOption(
    runtime.MIMEWildcard, &runtime.JSONPb{EmitDefaults: true}))
```

(Personally, I tend to pick the first.)